posts/2024-07-30-liberty-on-amazon-ecs.adoc

```diff
@@ -0,0 +1,265 @@
+
+For this blog, as the application is not CPU or memory intensive, we will use the amount of requests hitting the ALB we receive to trigger scaling events.
+
+To create an ALB request Scaling policy, you can edit the instances scaling policy. To do this, once more revisit your cluster and select your Service. Then select `Update service` and set the `Desired tasks` to `1`. Expand `Service auto scaling` and you'll be presented with a form like the one in the diagram below. In this form:
+
+. Set the minimum number of tasks to `1` and the maximum to `2`
+. Click `+ Add scaling policy`
+. Give your policy a name e.g. `mp-sp`
+. Set the `ECS service metric` to `ALBRequestCountPerTarget`
+. Set the `Target value` to `2`
+. Set both the `Scale *out* cooldown period` and `Scale *in* cooldown period` to `30`
+. Click `Update`
+
+The target value is set to a very low value so that it is easier to cause a scaling out alarm to trigger and create new instances. This value should be scoped to the requirements of the application and also that the amount of other resources provided are capable of handling that type of workload.
+
+image::../img/blog/amazon_ecs_scaling_policy.png[Amazon ECS scaling policy, width=70%,align="center"]
+
+Having created our policy we can now test it by attempting to trigger the alarm and cause our service to increase the number of instances available. As our alarm is focused on requests against the ALB, to test it, we should invoke our applications URL to generate some traffic. Given our alarm requires 3 datapoints above our target threshold in a given period, we just need to invoke it 3 times in the period being measured. After the trigger has been activated you will see the number of instances scale out, showing that autoscaling is working as expected.
+
+image::/img/blog/amazon_ecs_scaled_instances.png[Amazon ECS scaled out service,width=90%,align="center"]
+
+You have now run and scaled your own MicroProfile Application on Amazon ECS with AWS Fargate!
+
+== Summary:
+
+Through this blog you've gained an understanding of the steps required to take a MicroProfile application running with Open Liberty and run and effectively scale it with Amazon ECS with AWS Fargate. Continue your learning by checking out some of the additional resources listed below.
+
+== Additional Resources
+* https://aws.amazon.com/ecs/[Amazon Elastic Container Service]
+* https://aws.amazon.com/fargate/[AWS Fargate]
+
+* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html[Amazon ECS Task Definitions]
+* https://docs.aws.amazon.com/AmazonECS/latest/bestpracticesguide/intro.html[Amazon ECS Best Practices]
+* https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html[Amazon Elastic Load Balancing]
```

Comment:
Given the cut down of the blog the follow links are now no longer relevant
- Amazon Elastic Load Balancing 
- Create an HTTPS listener for your Application Load Balancer
- AWS Permissions

While ALBs are referenced and created in the post - it is no longer to the depth that having the link is needed for as it is done within the definition of ECS